producer/event_producer.py

```diff
@@ -118,12 +140,63 @@ def msg_processor(self, msg):
         msg.cursor.send_feedback(flush_lsn=msg.data_start)
         self.check_shutdown()
 
+    @cachetools.cached(cache={}, key=lambda self, relation_id: cachetools.keys.hashkey(relation_id))
```

Comment:
don't follow, what is expiry here ?
what happens when tablename changes - unlikely but still possible